tasks/WaypointNavigator.cpp

```diff
@@ -58,9 +58,9 @@ void WaypointNavigator::updateHook()
         
         delta.time = base::Time::now();
         delta.linear = wp.cmd.linear - pose.position;
-        delta.angular(0) = wp.cmd.angular(0) - base::getRoll(pose.orientation);
-        delta.angular(1) = wp.cmd.angular(1) - base::getPitch(pose.orientation);
-        delta.angular(2) = wp.cmd.angular(2) - base::getYaw(pose.orientation);
+        delta.angular(0) = base::Angle::normalizeRad(wp.cmd.angular(0) - base::getRoll(pose.orientation));
+        delta.angular(1) = base::Angle::normalizeRad(wp.cmd.angular(1) - base::getPitch(pose.orientation));
+        delta.angular(2) = base::Angle::normalizeRad(wp.cmd.angular(2) - base::getYaw(pose.orientation));
```

Comment:
One more thing. getRoll, getPitch and getYaw all call getEuler and then return only the needed element. For efficiency lets call getEuler once directly.